File: students/students/src/MON.Models/StudentModels/StudentAwardModel.cs
namespace MON.Models.StudentModels
{
    using System;
    using System.Collections.Generic;

    public class StudentAwardModel
    {
        public int? Id { get; set; }
        public DateTime Date { get; set; }
        public int PersonId { get; set; }
        public int AwardTypeId { get; set; }
        public string Description { get; set; }
        public string OrderNumber { get; set; }
        public string AdditionalInformation { get; set; }
        public int? InstitutionId { get; set; }
        public int AwardCategoryId { get; set; }
        public int FounderId { get; set; }
        public int AwardReasonId { get; set; }
        public short? SchoolYear { get; set; }
        public string SchoolYearName { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }
        public string InstitutionName { get; set; }
        public string AwardTypeName { get; set; }
        public string AwardCategoryName { get; set; }
        public string AwardReasonName { get; set; }
        public string FounderName { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/studentAwardModel.js
import Constants from '@/common/constants.js';
import { DocumentModel } from '@/models/documentModel.js';

export class StudentAwardModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id || null;
    this.personId = obj.personId || null;
    this.date = (obj.date && moment) ? moment(obj.date).format(Constants.DATEPICKER_FORMAT) : '';
    this.orderNumber = obj.orderNumber || null;
    this.description = obj.description || null;
    this.awardTypeId = obj.awardTypeId || null;
    this.awardTypeName = obj.awardTypeName;
    this.institutionId = obj.institutionId || null;
    this.institutionName = obj.institutionName;
    this.awardCategoryId = obj.awardCategoryId || null;
    this.awardCategoryName = obj.awardCategoryName;
    this.founderId = obj.founderId || null;
    this.founderName = obj.founderName;
    this.awardReasonId = obj.awardReasonId || null;
    this.awardReasonName = obj.awardReasonName;
    this.additionalInformation = obj.additionalInformation || null;
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.schoolYear = obj.schoolYear || null;
    this.schoolYearName = obj.schoolYearName;
  }
}
